text/0192-remove-constructs-compat.md

```diff
@@ -0,0 +1,667 @@
+
+To address this, the "constructs" library allows customizing these keys through a settings module. We will need to add that configuration to the `App` level so this behavior will be preserved.
+
+Alternatively, we can consider also modifying the CLI to accept the new keys as
+well, but since these are weakly coupled, it may introduce unwanted breakage,
+without much need.
+
+We recommend the first approach.
+
+# Drawbacks
+
+## User migration effort
+
+The main drawback from users' point of view is the introduction of the
+aforementioned breaking changes as part of the transition to CDK 2.0. As
+mentioned above, for the majority of users, the migration will be trivial and
+mechanical (import from "constructs" instead of "@aws-cdk/core").
+
+The removal of the "prepare" and "synthesize" hooks may require users to rethink
+their design in very advanced scenarios. We will create a GitHub issue to
+consult users on alternative designs.
+
+## CDK codebase migration efforts
+
+The AWS CDK codebase itself utilizes all of these APIs, and the migration effort
+is quite substantial.
+
+Having said that, the majority of this work is already complete and a branch is
+being maintained with these changes as a pre-cursor to the v2.x fork.
+
+## Staging of the 2.x fork
+
+Since this change involves modifications to the CDK's source code, it may cause
+merge conflicts as during the period in which we need to forward-port or
+back-port code between the v1.x branch and the v2.x branches.
+
+The key would be to continuously merge between the branches.
+
+# Rationale and Alternatives
+
+As a general rule, software layers which do not provider value to users should
+not exist. The constructs compatibility layer was added as solution for
+maintaining backwards compatibility within the v1.x version line while we
+extract `constructs` into an independent library.
+
+Redundant layers are expensive to maintain and are prone to idiosyncrasies as
+they evolve over time (for example, a CDK engineer may be tempted to add an
+AWS-specific feature in this layer, making it harder to clean up later).
+
+If we consider the various [reasons](#drawbacks) not to take this change, the
+main reason would be to simplify the [migration for users from 1.x to
+2.x](#user-migration-effort). The major version 2.x is already required to
+introduce monolithic packaging, and this change, for most users, is likely to be
+trivial (see [above](#breaking-changes)). Therefore, we believe this is probably
+not the correct motivation to reject this proposal.
+
+The [repository migration](#repository-migration-efforts) efforts and
+[co-existence of 2.x/1.x](#co-existence-of-2x1x) are both one-off costs this
+proposal suggests ways to reduce the chance for merge conflicts across these
+branches.
+
+## Alternatives considered
+
+At a high-level, we may consider to postpone this change to v3.x or to never
+take it, leaving this compatibility layer in place for eternity.
```

Comment:
... and eternity gets pretty long towards the end :)

Reply:
:-)